File: 프로그래머스/2/42578. 의상/의상.py
'''
딕셔너리 만들어서 key는 의상 이름, value는 종류
value의 개수 + value당 key 하나씩 조합
부분 집합
'''
def solution(clothes):
    hash_dict = {}
    
    for cloths in clothes:
        if cloths[1] in hash_dict:
            hash_dict[cloths[1]] += [cloths[1]]
        else:
            hash_dict[cloths[1]] = [cloths[1]]
            
    answer = 1
    for _, cloths_name in hash_dict.items():
        answer *= len(cloths_name) + 1
        
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const clothesMap = {};
    for (const [name, type] of clothes) {
        if (!clothesMap[type]) clothesMap[type] = 1;
        else clothesMap[type]++;
    }
    let ans = 1;
    for (const key in clothesMap) ans *= (clothesMap[key]+1);
    
    return ans-1; 
}